litestar/dto/base_dto.py

```diff
@@ -227,7 +228,7 @@ def create_openapi_schema(
         key = "data_backend" if field_definition.name == "data" else "return_backend"
         backend = cls._dto_backends[handler_id][key]  # type: ignore[literal-required]
 
-        if backend.wrapper_attribute_name:
+        if backend.wrapper_attribute_name and not field_definition.is_subclass_of(Response):
```

Comment:
I'm not sure if this is the right part to fix, because, as this section indicates, we *do* supported wrapped return types, such as `Response`.

https://github.com/litestar-org/litestar/blob/af1af6ba0b528cba3e90c44b83421e75b681ce64/tests/unit/test_dto/test_factory/test_integration.py#L685-L714

If this is the same case and just a bug within that, could you amend the test case so the schema is checked within the test cases linked above? That makes it easier to track which behaviour belongs together logically

Reply:
I'm still not sure about the special casing of `Response`. This should work with all generic container types really. 

We do have `AbstractDTO.resolve_generic_wrapper_type`, which is supposed to do what you're special casing here, so I think we have a bug somewhere in that function, where it's not correctly detecting / unwrapping things

Reply:
Thanks for the thorough investigation @mmev! I think ultimately, we need to do some refactoring to get all the response special-casing out, but for now, this is a reasonable fix.